### PR TITLE
Create law and test: Van-der-Waals state equation

### PR DESCRIPTION
**Pull request by Voldoned**

### CODE REVIEW

File: symplyphysics/laws/thermodynamics/equations_of_state/van_der_vaals_state_equation.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+volume = Symbol("volume", units.volume)
+temperature = Symbol("temperature", units.temperature)
+amount_of_substance = Symbol("amount_of_substance", units.amount_of_substance)
+
+bonding_forces_parameter = Symbol("bonding_forces_parameter", units.pressure * units.volume ** 2)
+molecules_volume_parameter = Symbol("molecules_volume_parameter", units.volume)
+
+
+law = Eq(
+    (pressure - bonding_forces_parameter * (amount_of_substance / volume) ** 2) * (volume - bonding_forces_parameter * amount_of_substance),
```
Comment (blackyblack) [inline symplyphysics/laws/thermodynamics/equations_of_state/van_der_vaals_state_equation.py:34]: Molecules volume is unused.

File: test/thermodynamics/equations_of_state/van_der_vaals_state_equation_test.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    with raises(TypeError):
+        vaals_law.calculate_pressure(test_args.v, test_args.t, test_args.nu, test_args.a, 100)
+
+
+def test_bad_bonding_forces_parameter(test_args):
+    ab = Quantity(1 * units.coulomb)
+    with raises(errors.UnitsError):
+        vaals_law.calculate_pressure(test_args.v, test_args.t, test_args.nu, ab, test_args.b)
+    with raises(TypeError):
+        vaals_law.calculate_pressure(test_args.v, test_args.t, test_args.nu, 100, test_args.b)
```
Comment (blackyblack) [inline test/thermodynamics/equations_of_state/van_der_vaals_state_equation_test.py:73]: Add new line.

File: test/thermodynamics/equations_of_state/van_der_vaals_state_equation_test.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    b = Quantity(1.115 * 1E-6 * units.meters ** 3)
+    Args = namedtuple("Args", ["t", "v", "nu", "a", "b"])
+    return Args(t=t, v=v, nu=nu, a=a, b=b)
+
+
+def test_basic_pressure(test_args):
+    result = vaals_law.calculate_pressure(test_args.v, test_args.t, test_args.nu, test_args.a, test_args.b)
+    assert SI.get_dimension_system().equivalent_dims(result.dimension, units.pressure)
+    result_ = convert_to(result, units.atm).evalf(5)
+    assert result_ == approx(90, 0.01)
```
Comment (blackyblack) [inline test/thermodynamics/equations_of_state/van_der_vaals_state_equation_test.py:41]: You have two typos in law yet you've got correct answer. How is it possible?

File: test/thermodynamics/equations_of_state/van_der_waals_state_equation_test.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    b = Quantity(4.225 * 1E-5 * units.meters ** 3 / units.moles)
+    Args = namedtuple("Args", ["t", "v", "nu", "a", "b"])
+    return Args(t=t, v=v, nu=nu, a=a, b=b)
+
+
+def test_basic_pressure(test_args):
+    result = waals_law.calculate_pressure(test_args.v, test_args.t, test_args.nu, test_args.a, test_args.b)
+    assert SI.get_dimension_system().equivalent_dims(result.dimension, units.pressure)
+    result_ = convert_to(result, units.atm).evalf(5)
+    assert result_ == approx(88.8, 0.01)
```
Comment (blackyblack) [inline test/thermodynamics/equations_of_state/van_der_waals_state_equation_test.py:41]: It should be 90 atm, according to reference example.

File: test/thermodynamics/equations_of_state/van_der_waals_state_equation_test.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+# Pressure should be 90 atm
+
+
+@fixture(name="test_args")
+def test_args_fixture():
+    t = Quantity(300 * units.kelvins)
+    v = Quantity(0.25 * units.liters)
+    nu = Quantity(1 * units.mole)
+    a = Quantity(0.188 * units.pascals * (units.meter ** 3 / units.mole) ** 2)
+    b = Quantity(4.225 * 1E-5 * units.meters ** 3 / units.moles)
```
Comment (blackyblack) [inline test/thermodynamics/equations_of_state/van_der_waals_state_equation_test.py:32]: Dimensions do not correspond to description.

File: test/thermodynamics/equations_of_state/van_der_waals_state_equation_test.py (hunk 4 above)
Comment (Voldoned) [reply test/thermodynamics/equations_of_state/van_der_waals_state_equation_test.py]: This is due to the transfer of the dimension from Pa to atm. 1 atm = 1.013 * 10^5 Pa. I rounded it up to 10^5 Pa and it immediately affected the result. I have now recalculated it without rounding, which I did last time.

**Review by blackyblack — APPROVED** [review]

**Pull request merged**